src/scanner/scanNFTStats.ts

```diff
@@ -0,0 +1,404 @@
+import {
+	trackAuctionData,
+	processAuctionSoldEvent,
+	trackAuctionBundleData,
+} from "./utils/trackTokenAuction";
+import { trackBurnBatchData, trackBurnData } from "./utils/trackTokenBurn";
+import { trackBuyData } from "./utils/trackTokenBuy";
+import {
+	trackAdditionalTokenData,
+	trackTokenSeriesData,
+	trackUniqueMintData,
+} from "./utils/trackTokenCreation";
+import { trackSeriesNameData } from "./utils/trackTokenName";
+import { trackSellBundleData, trackSellData } from "./utils/trackTokenSell";
+import {
+	trackTransferBatchData,
+	trackTransferData,
+} from "./utils/trackTokenTransfers";
+import { trackBidData } from "./utils/trackBidData";
+
+import { Api } from "@cennznet/api";
+import { config } from "dotenv";
+import { logger }  from "../logger";
+import { Vec } from "@polkadot/types-codec";
+import { u8aToString } from "@polkadot/util";
+import mongoose from 'mongoose';
+import {SignedBlock} from "@polkadot/types/interfaces/runtime";
+import {fetchNFTBlockForRange} from "./utils/fetchNFTBlockNumberForRange";
+import {updateProcessedBlockInDB} from "./dbOperations";
+const { LastBlockScan } = require('../mongo/models');
+config();
+export let supportedAssets = [];
+
+// find the event for the extrinsic
+function filterExtrinsicEvents(extrinsicIdx, events) {
+	return events.filter(
+		({ phase }) =>
+			phase.isApplyExtrinsic && phase.asApplyExtrinsic.eqn(extrinsicIdx)
+	);
+}
+
+function isExtrinsicSuccessful(extrinsicIdx, events) {
+	return (
+		events.findIndex((evt) => evt.event.method === "ExtrinsicSuccess") > -1
+	);
+}
+
+function getExtrinsicParams(e) {
+	return e.meta.args.map((arg, idx) => {
+		const value = e.args[idx].toJSON();
+		return {
+			...arg.toJSON(),
+			value,
+		};
+	});
+}
+
+function getTimestamp(block, api) {
+	for (const e of block.extrinsics) {
+		const call = api.findCall(e.callIndex);
+		if (call.section === "timestamp" && call.method === "set") {
+			const date = new Date(e.args[0].toJSON());
+			if (isNaN(date.getTime())) {
+				throw new Error("timestamp args type wrong");
+			}
+			return date;
+		}
+	}
+}
+
+async function processNFTExtrinsicData({
+	method,
+	params,
+	events,
+	txHash,
+	blockTimestamp,
+	api,
+	blockNumber,
+	owner,
+	blockHash,
+}) {
+	logger.info(`Event triggered::${method}`);
+	const date = blockTimestamp;
+	const findNFTEvent = events.find(({ event }) => event.section === "nft");
+	const eventData = findNFTEvent ? findNFTEvent.event.data.toJSON() : null;
+	switch (method) {
+		case "mintUnique": {
+			if (!eventData) {
+				logger.error(
+					`Something wrong, no event found for mintUnique extrinsic at blockNumber ${blockNumber}`
+				);
+				break;
+			}
+
+			await trackUniqueMintData(
+				eventData,
+				api,
+				params,
+				date,
+				owner,
+				txHash,
+				blockNumber
+			);
+			break;
+		}
+		case "mintSeries": {
+			if (!eventData) {
+				logger.error(
+					`Something wrong, no event found for mint series extrinsic at blockNumber ${blockNumber}`
```

Comment:
should be `minSeries` to match with the extrinsic method? can you check and update all of them please.

Reply:
mint series is better to read over mintSeries when debugging the log

Reply:
cool, just personal preferences, let's go with what you got